.maintain/gitlab/check_companion_build.sh

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env bash
+#
+# check if a pr is compatible with companion pr or master if not available
+#
+# to override one that was just mentioned mark companion pr in the body of the
+# pr like
+#
+# $REPO companion: $ORGANISATION/$REPO#567
+#
+# $ORGANISATION and $REPO are set using $1 and $2. You can also specify a custom
+# build command with $3
+# Every argument after $3 is for specifying *additional* dependencies this
+# project has that depend on substrate, which might also have companion PRs.
+
+# Example: Cumulus relies on both substrate and polkadot. If this substrate PR
+# requires a companion build on polkadot, when we are testing that cumulus builds
+# with this commit of substrate, we *also* need to clone & patch polkadot, and tell
+# cumulus to use this cloned+patched version, else the build is guaranteed to fail
+# (since it doesn't have the changes to polkadot that were required in the polkadot
+# companion PR)
+
+# So invoke this script like (arguments in [] indicate optional arguments)
+# ./check_companion_build.sh paritytech cumulus ['cargo test --release'] [paritytech/polkadot]
+
+set -e
+
+# Include the common functions library
+#shellcheck source=../common/lib.sh
+. "$(dirname "${0}")/../common/lib.sh"
+
+ORGANISATION=$1
+REPO=$2
+BUILDSTRING=${3:-cargo test --workspace --release}
+DEPS=("${@:4}")
+SUBSTRATE_DIR="$(pwd)"
+
+boldprint () { printf "|\n| \033[1m%s\033[0m\n|\n" "${@}"; }
+boldcat () { printf "|\n"; while read -r l; do printf "| \033[1m%s\033[0m\n" "${l}"; done; printf "|\n" ; }
+
+boldcat <<-EOT
+
+
+check_${REPO}_companion_build
+==============================
+
+this job checks if there is a string in the description of the pr like
+
+$REPO companion: $ORGANISATION/$REPO#567
+
+
+it will then run cargo check from this ${REPO}'s branch with substrate code
+from this pull request. otherwise, it will uses master instead
+
+
+EOT
+
+# Set the user name and email to make merging work
+git config --global user.name 'CI system'
+git config --global user.email '<>'
+
+# Merge master into our branch before building to make sure we don't miss
+# any commits that are required.
+git fetch --depth 100 origin
+git merge origin/master
```

Comment:
can it just be `git pull origin master`? I don't quite get the depth numbers but fetching a single branch is better than doing `fetch origin` (which fetches everything)